packages/stepper/src/stories/Stepper.components.tsx

```diff
@@ -0,0 +1,212 @@
+							: rhf.handleSubmit(() => alert('You did it!'))()
+					}
+				/>
+			</Form>
+		</FormProvider>
+	);
+};
+
+export const FormComponentStep2 = () => {
+	const { onDisableStep, onEnableStep, onNext, onPrevious, steps, currentStep } = useContext(
+		StepperForm.Context,
+	);
+	const { stepsData, setStepsData } = useContext(BusinessContext);
+
+	const currentStepNavigation = steps[currentStep].navigation;
+
+	const rhf = useForm<{ randomInput: string }>({
+		defaultValues: { ...stepsData[1] },
+	});
+
+	const { field, meta } = useController({
+		control: rhf.control,
+		name: 'randomInput',
+		rules: {
+			required: true,
+		},
+	});
+
+	useEffect(() => {
+		const data = stepsData;
+		data[1] = { [field.name]: field.value };
+		setStepsData(data);
+	}, [field]);
+
+	return (
+		<FormProvider {...rhf}>
+			<Form>
+				<Form.Fieldset>
+					<Form.Checkbox
+						id="disable1"
+						name="disable-steps"
+						label="Disable step 3"
+						onChange={e => {
+							if (e.target.checked) {
+								onDisableStep('STEP3', 'Just because');
+							} else {
+								onEnableStep('STEP3');
+							}
+						}}
+					/>
+
+					{meta.invalid && (
+						<InlineMessageDestructive description={'This field is required'} withBackground />
+					)}
+					<Form.Input
+						id="randomInput"
+						label="Random input"
+						name={field.name}
+						onChange={field.onChange}
+						value={field.value}
+					/>
+				</Form.Fieldset>
+				<StepperForm.Footer
+					onCancel={() => alert('Cancel action')}
+					onNext={() =>
+						currentStepNavigation?.next
+							? rhf.handleSubmit(onNext)()
+							: rhf.handleSubmit(() => alert('You did it!'))()
+					}
+					onPrevious={onPrevious}
+				/>
+			</Form>
+		</FormProvider>
+	);
+};
+
+export const FormComponentStep3 = () => {
+	const { onNext, onPrevious, steps, currentStep } = useContext(StepperForm.Context);
+	const { stepsData, setStepsData } = useContext(BusinessContext);
+
+	const currentStepNavigation = steps[currentStep].navigation;
+
+	const rhf = useForm<{ randomInput: string }>({
+		defaultValues: { ...stepsData[2] },
+	});
+
+	const { field, meta } = useController({
+		control: rhf.control,
+		name: 'randomInput',
+		rules: {
+			required: true,
+		},
+	});
+
+	useEffect(() => {
+		const data = stepsData;
+		data[2] = { [field.name]: field.value };
+		setStepsData(data);
+	}, [field]);
+
+	return (
+		<FormProvider {...rhf}>
+			<Form>
+				<Form.Fieldset>
+					{meta.invalid && (
+						<InlineMessageDestructive description={'This field is required'} withBackground />
+					)}
+					<Form.Input
+						id="randomInput"
+						label="Random input"
+						name={field.name}
+						onChange={field.onChange}
+						value={field.value}
+					/>
+				</Form.Fieldset>
+				<StepperForm.Footer
+					onCancel={() => alert('Cancel action')}
+					onNext={() =>
+						currentStepNavigation?.next
+							? rhf.handleSubmit(onNext)()
+							: rhf.handleSubmit(() => alert('You did it!'))()
```

Comment:
This mechanism looks duplicated in all steps. Do you think `rhf.handleSubmit(onNext)()` could be called in any case and `onNext` function would be responsible to call either stepperForm.onNext or a new function passed to stepperForm like `onFinish` (or whatever you want to name it). Because it looks to be the "last" step. `onFinish` could be called with step key in order to add custom business logic if you want user to finish differently based on final step ?